docs/src/sections/webapp_solution.adoc

```diff
@@ -0,0 +1,529 @@
+* _Form_ — komponenty do tworzenia formularzy
+* _Layouts_ — komponenty, które wyświetlają inne komponenty dodając do nich pewne elementy
+* _Modals_ — okna dialogowe
+
+Reszta katalogów w folderze _components_ jest nazwana od widoków, gdzie są wykorzystywane:
+
+* auth
+* adminPanel
+* wallet
+* History
+* Quickbuy
+* Transactions
+
+==== Pliki konfiguracyjne
+
+* postcss.config.js — konfiguracja narzędzia _postcss_, wymagana przez _tailwindcss_
+* jest.config.js — zawiera konfiguracje takich rzeczy jak generowanie raportów z testów,
+czy dodawanie ścieżek modułów wykorzystywanych w testach
+* .prettierrc — konfiguracja narzędzia _prettier_
+* .tailwind.config.js — konfiguracja narzędzia _tailwind_
+* .vite.config.ts — zawiera ścieżki, których zapytania powinny być przekazane przez _vite_ do serwera,
+ponieważ w środowisku deweloperskim serwer i aplikacja internetowa są uruchomione na różnych portach
+[source,typescript]
+----
+include::../../../webapp/vite.config.ts[lines=7..20]
+----
+** .eslintrc — zawiera informacje o tym, jakiego typu problemy ze składnią powinien zgłaszać _eslint_
+przy analizowaniu kodu
+
+==== Nawigacja
+
+W jednostronicowych aplikacjach internetowych zmiana widoku nie powinna wywołać przeładowania strony.
+Ścieżka URL w przeglądarce musi być więc aktualizowana przez kod _JavaScript_.
+W tym celu została wykorzystana biblioteka _react-router-dom_, która wyświetla odpowiednie komponenty
+zależnie od ścieżki oraz udostępnia funkcje do jej aktualizowania.
+Przypisanie komponentów do ścieżek można zobaczyć w komponencie głównym aplikacji - _src/components/App.tsx_.
+
+.Fragment pliku App.tsx.
+[source, tsx]
+----
+include::../../../webapp/src/components/App.tsx[lines=40..57]
+...
+----
+
+Ścieżki URL aplikacji webowej zostały zapisane w postaci _JSON_ do pliku _src/routesBuilder.ts_
+w celu zwiększania elastyczności kodu.
+
+Oprócz zdefiniowanych ścieżek widać, że na samej górze drzewa komponentów znajdują się contexty i layouty.
+Layout _NavbarLayout.tsx_ zagnieżdża prawie każdy widok, ponieważ dodaje górny pasek nawigacyjny do widoków.
+Jedyne widoki, które owego paska nie posiadają, to _LogoutPage.tsx_ i _Error404Page.tsx_.
+Do tego Sprawdza, czy _UserContext.tsx_ wczytuje token, jeśli tak, to dodaje powiadomienie z informacją,
+że token jest w trakcie ładowania.
+
+.Fragment pliku NavbarLayout.tsx.
+[source, tsx]
+----
+include::../../../webapp/src/components/Layouts/NavbarLayout.tsx[lines=9..32]
+----
+
+.Wiadomość o wczytywaniu tokena w lewym dolnym rogu.
+image::../images/system_presentation/webapp/refreshing_token.png[]
+
+_UserLayout.tsx_, tak samo, jak _AdminLayout.tsx_ i _PublicLayout.tsx_ przekierowuje użytkownika
+w zależności od jego stanu.
+Zagnieżdżają w sobie komponenty w zależności od tego, komu należy się udostępnić dany widok.
+
+* _UserLayout.tsx_ — przekierowuje do strony logowania, jeśli użytkownik nie jest zalogowany, lub do strony głównej,
```

Comment:
przydałoby się zredagować to zdanie. możesz dać:
"... do strony logowania - jeśli .... lub(bez przecinka) do strony głównej - jeśli ..."